runtime/queries/metricsview.go

```diff
@@ -24,11 +26,14 @@ func lookupMetricsView(ctx context.Context, rt *runtime.Runtime, instanceID, nam
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	if obj.GetMetricsView() == nil {
+	if !obj.IsMetricsView() {
```

Comment:
`GetMetricsView()` panics if its not a metrics view so the check `obj.GetMetricsView() == nil` does not make sense as it will never go inside `if` condition. So added this method as its used during query cache key computation - if entry is metricsview then add user attr to the key, it was also being used in `column_time_range` where we don't want to panic if its not a metricsview but just move on.
```
func (e *CatalogEntry) GetMetricsView() *runtimev1.MetricsView {
	obj, ok := e.Object.(*runtimev1.MetricsView)
	if !ok {
		panic(fmt.Errorf("entry '%s' is not a metrics view", e.Name))
	}
	return obj
}
```